src/main/java/mezz/jei/plugins/vanilla/ingredients/fluid/FluidStackHelper.java

```diff
@@ -49,6 +58,18 @@ public String getUniqueId(FluidStack ingredient) {
 		return "fluid:" + registryName;
 	}
 
+	@Override
+	public String getUniqueId(FluidStack ingredient, UidContext context) {
```

Comment:
These changes should be done on `public String getUniqueId(FluidStack ingredient)` instead. We don't use `context` in here so we can leave it with the default implementation, that way both methods will use the subtype info.